src/main/java/com/google/devtools/build/lib/bazel/commands/ModCommand.java

```diff
@@ -475,6 +476,7 @@ private BlazeCommandResult execInternal(CommandEnvironment env, OptionsParsingRe
     }
 
     ImmutableMap<String, RepoDefinition> targetRepoDefinitions = null;
+    String repoOverrideMsg = "";
```

Comment:
this is a rather awkward way to show a message. In particular, it wouldn't work if there were multiple overridden repos requested (like `bazel mod show_repo @bazel_tools @some_other_overridden_repo` -- we'd only show the last one).

Instead, we should make `targetRepoDefinitions` an `ImmutableMap<String, RepoDefinitionValue>`, so that we can store `RepoDefinitionValue.RepoOverride` in there too. Then in the `for` loop at ModExecutor.java:165, we can then switch between printing a `RepoDefinition` object (if `RepoDefinitionValue.Found`) or a special message (if `RepoDefinitionValue.RepoOverride`).